src/Resources/config/menu.xml

```diff
@@ -49,7 +38,7 @@
             <tag name="knp_menu.provider" />
         </service>
 
-        <service id="knp_menu.menu_provider.container_aware" class="%knp_menu.menu_provider.container_aware.class%" public="false">
+        <service id="knp_menu.menu_provider.container_aware" class="Knp\Menu\Provider\PsrProvider" public="false">
```

Comment:
The service definition needs to be dropped, as the compiler pass is removing it anyway on modern Symfony versions using LazyProvider (or at least, it was before this PR).

Reply:
Are you sure? I'm trying this in a project of mine (with Symfony 3.4) and I can see `knp_menu.menu_provider.container_aware` service in `debug:container`

Reply:
are you trying your own branch, or the existing master branch ? Your own branch does not remove it anymore, because you deleted the line removing the definition.

Reply:
So, if I understand correctly, we should remove the whole `MenuPass` too

Reply:
@stof can we also remove `container_aware` option?

Reply:
yes. MenuPass should dropped in favor of RegisterMenuPass

And the option should also be dropped (deprecating it in 2.x first)